src/custom-components/federated-search/components/FedFilter.vue

```diff
@@ -0,0 +1,218 @@
+            >
+                Scroll right
+            </VsButton>
+        </div>
+    </div>
+</template>
+
+<script setup>
+// Buttons should be links on homepage!
+import VsButton from '@/components/button/Button.vue';
+import VsBody from '@/components/body/Body.vue';
+
+/**
+ * Fed Filter is used in the Federated Search engine.
+ * Top level variant to pick between broad categories,
+ * Second level variant for fine grain filtering.
+ *
+ * @displayName FedFilter
+ */
+
+const props = defineProps({
+    /*
+    * Array of filter options
+    * filterCategories: {
+    *   id: string,
+    *   label: string,
+    *   icon?: string,
+    * }
+    */
+    filterCategories: {
+        type: Array,
+        required: true,
+    },
+    /**
+     * Variant of filter:
+     * Primary: medium buttons with icons and 1em gap
+     * Secondary: small buttons, no icons and 0.5em gap
+     */
+    variant: {
+        type: String,
+        default: 'primary',
+        validator: (value) => value.match(
+            /(primary|secondary)/,
+        ),
+    },
+    /* Used to determine if the filter will wrap
+     * rather than scroll on desktop or not
+     */
+    wrap: {
+        type: Boolean,
+        default: false,
+    },
+    /**
+     * ID of currently active filter
+     */
+    activeFilter: {
+        type: [String, Array],
+        default: '',
+    },
+    /**
+     * Enables the scroll buttons on filter
+     */
+    scrollButtons: {
+        type: Boolean,
+        default: false,
+    },
+    /**
+     * Tells if JS is disabled
+     */
+    jsDisabled: {
+        type: Boolean,
+        default: false,
+    },
+});
+
+defineEmits(['filter-updated']);
+
+function isActive(category) {
+    if (typeof props.activeFilter === 'string') {
+        return props.activeFilter === category;
+    }
+
+    if (props.activeFilter && typeof props.activeFilter === 'object') {
+        return props.activeFilter.includes(category);
+    }
+
+    return false;
+}
+/**
+ * This is not the best solution by any means
+ * There is a better solution that doesn't rely on JS,
+ * but is not stable across all browsers yet.
+ * We should look to implent it, if we chose to keep it,
+ * when it does become stable on Safari & Firefox.
+ * https://developer.mozilla.org/en-US/docs/Web/CSS/::scroll-button
+*/
+function scroll(dir) {
+    const scrollRail = document.getElementById(`vs-fed-filter--scroll-rail_${props.variant}`);
+
+    if (dir === 'left') scrollRail?.scrollBy(-200, 0);
+    else if (dir === 'right') scrollRail?.scrollBy(200, 0);
+}
+
+function filterClasses() {
+    return [
+        `vs-fed-filter--scroll-rail_${props.variant}`,
+        props.wrap ? 'vs-fed-filter--scroll-rail_wrap' : '',
+    ];
+}
+</script>
+
+<style lang="scss">
+    .vs-fed-filter {
+        &--header {
+            font-weight: $vs-font-weight-medium;
+        }
+
+        &--scroll-container{
+            display: flex;
+        }
+
+        &--scroll-rail {
```

Comment:
For consistency, the scrollbar should look the same as the card group styles - might be best to take the scrol; snap webkit styles from src\components\card-group\CardGroup.vue and put them in a _placeholders.scss for use across the site?